### PR TITLE
sw: Add scalafmt configuration file

### PR DESCRIPTION
**Pull request by xiaoling-yi**
Add scalafmt autoformatter for really  cool style

### CODE REVIEW

**Review by JosseVanDelm — CHANGES_REQUESTED** [review]
Add github action for scalafmt

**Comment by xiaoling-yi** [comment]
Hi Josse, the stb-scalafmt has been added successfully.

**Review by JosseVanDelm — CHANGES_REQUESTED** [review]
Hi @xiaoling-yi,

The tests you setup are not running correctly.
I suggest the following:
* remove the actual SBT test and firtool installation from this PR, let's just focus on formatting instead
* Why do we need to test for java 8, 11, and 17? I think pinning to one version should be okay :smile: , at least for now!
* Are you sure you are correctly checking out the repo with fetch-depth=0? Otherwise the format tests might return a false positive!

File: .github/workflows/unit_test.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Unit Test
+on: [push, pull_request]
+  
+jobs: 
+  run:
+    name: Build and Run
+    strategy:
+      matrix:
+        java-version: [8, 11, 17]
```
[the remark this answers is not in the document]
Comment (xiaoling-yi) [reply .github/workflows/unit_test.yml]: I deleted version 8 and 17.

File: .github/workflows/unit_test.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout current branch (full)
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```
[the remark this answers is not in the document]
Comment (xiaoling-yi) [reply .github/workflows/unit_test.yml]: Oh, I have changed fetch-depth to 1.

File: .github/workflows/unit_test.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        run: wget -q -O - https://github.com/llvm/circt/releases/download/firtool-1.42.0/firrtl-bin-ubuntu-20.04.tar.gz | tar -zx | echo "$(pwd)/firrtl-bin-ubuntu-20.04/firtool-1.42.0/bin" >> $GITHUB_PATH 
+
+      - name: Check PATH
+        run: echo $GITHUB_PATH
+
+      - name: Check Firtool version
+        run: firtool -version
+
+      - name: Run the unit tests
+        run: sbt test
```
[the remark this answers is not in the document]
Comment (xiaoling-yi) [reply .github/workflows/unit_test.yml]: Ok. no problem.

**Review by JosseVanDelm — APPROVED** [review]
This looks much better!

@xiaoling-yi why are we using version 11 specifically? Is this the version that works best for chisel?

**Comment by xiaoling-yi** [comment]
version 17 also works. It is compatible with our Scala and Chisel versions.

**Pull request merged**